src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -51,6 +52,7 @@ export const SpawnHostModal: React.VFC<SpawnHostModalProps> = ({
   visible,
   onCancel,
 }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
```

Comment:
We use [`useUpdateURLQueryParams`](https://github.com/evergreen-ci/spruce/blob/5f95e9396419a573b210db72d561b6d89fc41a46/src/hooks/useUpdateURLQueryParams.ts#L7) everywhere else to update query params it avoids the filtering you are doing below.